.semaphore/semaphore.yml

```diff
@@ -0,0 +1,725 @@
+            - |
+              for p in $(ls things/policies/*.pb.go); do
+                if ! cmp -s $p $p.tmp; then
+                  echo "Proto file and generated Go file $p are out of sync!"
+                  exit 1
+                fi
+              done
+            - |                
+              for p in $(ls pkg/messaging/*.pb.go); do
+                if ! cmp -s $p $p.tmp; then
+                  echo "Proto file and generated Go file $p are out of sync!"
+                  exit 1
+                fi
+              done
+  - name: Lint and Test Users
+    run:
+      when: "change_in(['users', 'cmd/users', 'users/policies/auth.proto'])"
+    dependencies:
+      - Setup
+      - Check Generated Protocol Buffer Files
+    task:
+      jobs:
+        - name: Lint Users
+          commands:
+            - cd users
+            - cache restore linter
+            - './bin/golangci-lint run --no-config --disable-all --enable gosimple,errcheck,govet,unused,goconst,godot --timeout 3m'
+        - name: Test Users
+          commands:
+            - cd users
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+      secrets:
+        - name: CODECOV_TOKEN
+  - name: Build Users
+    dependencies:
+      - Lint and Test Users
+    run:
+      when: "change_in(['users', 'cmd/users'])"
+    task:
+      jobs:
+        - name: Build Users
+          commands:
+            - make docker_users
+      secrets:
+        - name: DOCKER_TOKEN            
+        - name: DOCKER_USERNAME                    
+  - name: Lint and Test Things
+    run:
+      when: "change_in(['things', 'cmd/things', 'users/policies/auth.proto'])"
+    dependencies:
+      - Setup
+      - Check Generated Protocol Buffer Files
+    task:
+      jobs:
+        - name: Lint Things
+          commands:
+            - cd things
+            - cache restore linter
+            - './bin/golangci-lint run --no-config --disable-all --enable gosimple,errcheck,govet,unused,goconst,godot --timeout 3m'
+        - name: Test Things
+          commands:
+            - cd things
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+      secrets:
+        - name: CODECOV_TOKEN
+  - name: Build Things
+    dependencies:
+      - Lint and Test Things
+    run:
+      when: "change_in(['things', 'cmd/things'])"
+    task:
+      jobs:
+        - name: Build Things
+          commands:
+            - make docker_things
+      secrets:
+        - name: DOCKER_TOKEN            
+        - name: DOCKER_USERNAME              
+  - name: Lint and Test CoAP
+    run:
+      when: "change_in(['coap', 'cmd/coap', 'pkg/messaging/message.proto', 'things/policies/postgres', 'things/policies/service.go'])"
```

Comment:
```suggestion
      when: "change_in(['coap', 'cmd/coap',  'things/policies/postgres/policies.go', 'things/policies/api/grpc/client.go', 'things/policies/service.go'])"
```

Reply:
Do the same for other adapters and services using Things gRPC API.

Reply:
The same pattern goes for services that use Users gRPC API (`'users/policies/postgres/policies.go', 'users/policies/api/grpc/client.go', 'users/policies/service.go'`) so update those, as well.